interceptors/retry/backoff.go

```diff
@@ -16,29 +15,40 @@ func BackoffLinear(waitBetween time.Duration) BackoffFunc {
 	}
 }
 
+// JitterUp adds random jitter to the duration.
+// This adds or subtracts time from the duration within a given jitter fraction.
+// For example for 10s and jitter 0.1, it will return a time within [9s, 11s])
+func JitterUp(duration time.Duration, jitter float64) time.Duration {
+	multiplier := jitter * (rand.Float64()*2 - 1)
+	return time.Duration(float64(duration) * (1 + multiplier))
+}
+
+// ExponentBase2 computes 2^(a-1) where a >= 1. If a is 0, the result is 0.
+func ExponentBase2(a uint) uint {
+	return (1 << a) >> 1
+}
```

Comment:
Probably don't want these to be exported, in case we want to be able to change them.

Reply:
JitterUp is used by the package `retry_test` in `retry_test.go` file. Since test files cannot be in the same package - https://stackoverflow.com/questions/19998250/proper-package-naming-for-testing-with-the-go-language - I decided to keep this name public. Any suggestions to bypass this?

Reply:
I have made exponentbase2 private though

Reply:
Tests can definitely be in the same package. There's a difference between external tests and internal tests. Just make the tests internal (use the same package name as the package) and make this function unexported.

Reply:
Sure, this is a nice idea. I have converted the `retry_test` package into `retry` package.